### PR TITLE
mimir: Calculate stop_area weight by stop_point count

### PR DESCRIPTION
**Pull request by kadhikari**
This modification calculates weight of each StopArea which is the number of it's StopPoint.

### CODE REVIEW

File: src/bin/stops2mimir.rs [hunk 1]
```diff
@@ -127,12 +136,19 @@ impl<'a, R: std::io::Read + 'a> Iterator for StopPointIter<'a, R> {
                 Ok(())
             }
         }
+        
         self.iter.next().map(|r| {
             r.and_then(|r| {
                 let location_type = self.get_location_type(&r);
                 let visible = self.get_visible(&r);
+                let parent_station = self.get_parent_station(&r);
+                if let (Some(0), Some(id)) = (location_type, parent_station) {
```
Comment (antoine-de) [inline src/bin/stops2mimir.rs]: can you add a comment explaining this ?

File: src/bin/stops2mimir.rs [hunk 2]
```diff
@@ -194,6 +210,15 @@ fn attach_stops_to_admins<'a, It: Iterator<Item = &'a mut mimir::Stop>>(stops: I
           nb_unmatched,
           nb_matched + nb_unmatched);
 }
+                                                                        
+//Update weight value for each stop_area from HashMap.
+fn finalize_stop_area_weight<'a, It: Iterator<Item = &'a mut mimir::Stop>>(stops: It, nb_stop_points: &HashMap<String, u32>){
```
Comment (antoine-de) [inline src/bin/stops2mimir.rs]: you need a space before the `{`

**Comment by kadhikari** [comment]
jenkins test this please

File: src/bin/stops2mimir.rs [hunk 3]
```diff
@@ -127,12 +138,21 @@ impl<'a, R: std::io::Read + 'a> Iterator for StopPointIter<'a, R> {
...
+
         self.iter.next().map(|r| {
             r.and_then(|r| {
                 let location_type = self.get_location_type(&r);
                 let visible = self.get_visible(&r);
+                let parent_station = self.get_parent_station(&r);
+                // For each stop_point with location_type = 0 and parent_type not empty
+                // insert stop_area code in the HashMap with value = 0 if absent
+                // and increment the weight by one
+                if let (Some(0), Some(id)) = (location_type, parent_station) {
```
Comment (antoine-de) [inline src/bin/stops2mimir.rs]: hum my point was more that location_type=0 => it's a stoppoint is not straightforward.

maybe a comment like:

```rust
if let (Some(0), Some(sa_id)) = (location_type, parent_station) {
    // if it's a stop, we update its stop_area counter
    if !id.is_empty() {
        *self.nb_stop_points.entry(format!("stop_area:{}", sa_id)).or_insert(0) += 1;
    }
}
```

File: tests/bragi_stops_test.rs [hunk 4]
```diff
@@ -158,3 +159,18 @@ fn stop_all_data_test(bragi: &BragiHandler) {
...
+    //always be second.
+    let response = bragi.get("/autocomplete?q=weight&_all_data=true");
+    assert_eq!(response.len(), 2);
+
+    let stop = response.first().unwrap();
+
+    assert_eq!(get_value(stop, "type"), "public_transport:stop_area");
+    assert_eq!(get_value(stop, "label"), "weight three");
+    assert_eq!(get_value(stop, "name"), "weight three");
+    assert_eq!(get_value(stop, "id"), "stop_area:SA:weight_3_station");
```
Comment (antoine-de) [inline tests/bragi_stops_test.rs]: maybe check that the second is stop_area:SA:weight_1_station ?

File: src/bin/stops2mimir.rs (hunk 3 above)
Comment (TeXitoi) [inline src/bin/stops2mimir.rs]: If it's a stop **point**

Comment (antoine-de) [inline src/bin/stops2mimir.rs]: oups yes, that was the whole point of this comment :rofl:

File: tests/bragi_stops_test.rs (hunk 4 above)
Comment (kadhikari) [inline tests/bragi_stops_test.rs]: done

**Pull request merged**